### ISSUE TITLE
What's the best way to consume casadi from cmake for c++?

### ISSUE THREAD

**Issue by eike-fokken** [casadi/casadi#3222]
Hi!

What is the best way to build and use casadi for a c++ project without installing it to the system?

Within
[https://github.com/eike-fokken/consumecasadi.git
](https://github.com/eike-fokken/consumecasadi.git
)

I build casadi myself and provide a library target to be consumed. That git repository can be used as a git submodule. This solution needs more work to for example also provide all the plugins. But before I go that route I wanted to know, whether I just failed to see the official way.

**Comment by eike-fokken** [comment]
Ah, coming back to this I see an obvious question: Why not just use `add_subdirectory` in CMake.
This works great for casadi core but as soon as you want to load a plugin, it isn't found.

A less important thing: Casadi only works with make and I like ninja better. Nevertheless, how do you include casadi in a C++ project or do you always install it to the system?

**Comment by adarshrs** [comment]
Hi, did you find a solution to this that lets you use plugins?

**Comment by eike-fokken** [comment]
No, I didn't follow up on this as I then decided to build the prototype, I was going for, in python first: https://github.com/eike-fokken/pyconstrainedpacker
